src/sentry/web/frontend/setup_wizard.py

```diff
@@ -15,11 +22,29 @@
     ProjectKeyStatus,
     ProjectStatus,
 )
+from sentry.utils.http import absolute_uri
+from sentry.utils.urls import add_params_to_url
 from sentry.web.frontend.base import BaseView
 from sentry.web.helpers import render_to_response
 
 
 class SetupWizardView(BaseView):
+    def handle_auth_required(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
+        if request.GET.get("signup") == "1" and settings.SENTRY_SIGNUP_URL:
+
+            fragments = list(urlparse(absolute_uri(request.get_full_path())))
+
+            # get the params from the url and apply it to the signup url
+            params_for_signup = dict(parse_qsl(fragments[4]))
+            # remove the signup query param
+            params_for_signup.pop("signup", None)
+            # remove query params from next url
+            fragments[4] = ""
```

Comment:
Maybe a dumb question since I am lacking a lot of context: Is there a possibility we remove params we want to keep?

Reply:
@lforst I thought about it, but we likely want to apply the query parameters just to the `signup` URL. If this changes we can modify the implementation but as of yet we don't need it.